File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        int n = nums1.size();
        int m = nums2.size();
        vector<int>result;

        sort(nums1.begin(),nums1.end());
        sort(nums2.begin(),nums2.end());

        int i = 0;
        int j = 0;

        while(i<n && j<m){
            if(i > 0 && nums1[i] == nums1[i-1]) {
                i++;
                continue;
            }
           
            if(j > 0 && nums2[j] == nums2[j-1]) {
                j++;
                continue;
            }
            if(nums1[i] == nums2[j]){
                result.push_back(nums1[i]);
                i++;
                j++;
            }else if(nums1[i] < nums2[j]){
                i++;
            }else{
                j++;
            }
        }
        return result;
    }
           
            
             
            
        
      
};

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        int[] seen = new int[1001];

        for(int x : nums1){
            seen[x] = 1;
        }
        int k = 0;
        int[] arr = new int[nums2.length];
        for(int x : nums2){
            if(seen[x] == 1){
                arr[k++] = x;
                seen[x] = 0;
            }
        }
        return Arrays.copyOf(arr, k);
    }
}